src/components/Calculator.js

```diff
@@ -1,12 +1,58 @@
 import React, { useState } from 'react';
 import './Calculator.css';
 import Button from './Button';
+import calculate from '../logic/calculate';
+import operate from '../logic/operate';
 
 const Calculator = () => {
   const [display, setDisplay] = useState('0');
+  const [prevInput, setPrevInput] = useState(null);
+  const [currentInput, setCurrentInput] = useState(null);
+  const [operation, setOperation] = useState(null);
 
   const handleButtonClick = (value) => {
-    setDisplay(value);
+    if (/\d/.test(value)) {
+      if (!currentInput) {
+        setCurrentInput(value);
+        setDisplay(value);
+      } else {
+        setCurrentInput(currentInput + value);
+        setDisplay(currentInput + value);
+      }
+    } else if (value === '+' || value === '-' || value === 'x' || value === '÷') {
+      if (currentInput) {
+        if (prevInput && operation) {
+          const result = calculate(prevInput, currentInput, operation);
+          setPrevInput(result);
+          setCurrentInput(null);
+          setOperation(value);
+          setDisplay(result);
+        } else {
+          setPrevInput(currentInput);
+          setCurrentInput(null);
+          setOperation(value);
+        }
+      }
+    } else if (value === '=') {
+      if (prevInput && currentInput && operation) {
```

Comment:
- [ ] These logics have already been implemented in the original logic files, just as I have mentioned above,study the files and understand how to use them, you do not need these if else statements here in your component. :+1: :100: